src/components/svgSetting/svgSetting.js

```diff
@@ -1,34 +1,86 @@
-import React, { Component } from 'react';
-import '../checkBoxSelection/checkBoxSelection.scss';
-import PropTypes from 'prop-types';
+import React, { Component } from "react";
+import "../checkBoxSelection/checkBoxSelection.scss";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { Svgo } from "../svgSetting/svgoConfig";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../../store/actionTypes";
+
 class SvgSetting extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            isClicked: this.props.isSelected
-        };
+  constructor(props) {
+    super(props);
+    this.state = {
+      isClicked: this.props.isSelected,
+      arr: [],
+      svgoObject: { ...Svgo },
+      IndexOf: -1
+    };
+  }
+
+  handleSettingsClick = () => {
+    this.setState({
+      isClicked: !this.state.isClicked
+    });
+
+    this.setState({
+      IndexOf: this.props.svgOptions.findIndex(this.props.value)
+    });
+
+    if (this.state.IndexOf < 0) {
+      this.props.svgOptions.push(this.props.value);
+    } else {
+      this.props.svgOptions.splice(this.state.IndexOf, 1);
```

Comment:
splice mutates the original array please use a non mutation method instead

Reply:
I will use slice() instead then as it is not mutating.

Reply:
cool, good choice